### PR TITLE
[1LP][RFR] Added new test for XA Datasource creating. Adding new smoke test for non XA datasource creation.

### PR DESCRIPTION
**Pull request by hhovsepy**
{{pytest: cfme/tests/middleware/test_middleware_datasource.py -v --use-provider hawkular}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 7184f1abf5c7b3763c2a3366258c66868f273b76:

`cfme/tests/middleware/datasource_methods.py`:
- :bangbang: Line 67:101: **E501** *line too long (106 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by izapolsk** [comment]
it seems 57 PRT tests didn't manage to get appliance. I've restarted tests

File: cfme/tests/middleware/test_middleware_datasource.py [hunk 1]
```diff
@@ -131,6 +132,53 @@ def test_datasource_details(provider):
         ds_mgmt.validate_properties()
 
 
+@pytest.mark.meta(blockers=[BZ(1427141, forced_streams=["5.8", "upstream"])])
+@pytest.mark.uncollectif(lambda: current_version() < '5.8')
```
Comment (izapolsk) [inline cfme/tests/middleware/test_middleware_datasource.py]: REQUIRED: please use appliance.version() instead. current_version is deprecated.

**Comment by hhovsepy** [comment]
is it possible to restart tests for particular appliance only?

File: cfme/middleware/provider/__init__.py [hunk 2]
```diff
@@ -401,9 +404,14 @@ def add_datasource(self, ds_type, ds_name, jndi_name, driver_name,
         """
         self.load_details(refresh=True)
         datasources_btn("Add Datasource", invokes_alert=True)
+        if version.current_version() >= '5.8':
```
Comment (izapolsk) [inline cfme/middleware/provider/__init__.py]: REQUIRED: please don't use current_version. here you can use self.appliance.version() instead

**Comment by izapolsk** [comment]
I guess yes, via api or via ssh but it would be better to ask @mfalesni.

File: cfme/tests/middleware/test_middleware_datasource.py (hunk 1 above)
Comment (hhovsepy) [inline cfme/tests/middleware/test_middleware_datasource.py]: can not find any example of using "appliance.version()" in pytest.mark.uncollectif, can you please let me know how to do it?

Comment (izapolsk) [inline cfme/tests/middleware/test_middleware_datasource.py]: oh yes, sorry, my bad. appliance is fixture. it won't be accessible in uncollect.

**Comment by hhovsepy** [comment]
downstream-57z failure is because of simultaneous run on several versions,
upstream did not start

**Comment by hhovsepy** [comment]
@izapolsk @psav can this PR be merged?

File: cfme/tests/middleware/test_middleware_datasource.py (hunk 1 above)
Comment (mfalesni) [inline cfme/tests/middleware/test_middleware_datasource.py]: We should probably work out an uncollectif specialized on versions so we do not have to care about this stuff.

**Pull request merged**